jinja2/filters.py

```diff
@@ -518,9 +518,17 @@ def do_int(value, default=0, base=10):
     can also override the default base (10) in the second
     parameter, which handles input with prefixes such as
     0b, 0o and 0x for bases 2, 8 and 16 respectively.
+    The base is ignored for non-string values.
     """
+    # only string types support int() with explicit base
+    if isinstance(value, string_types) and base != 10:
+        try:
+            return int(value, base)
+        except ValueError:
+            return default
```

Comment:
On first though this is inconsistent, as we return the `default` in all other scenarios where the input cannot be parsed. But on the other hand there doesn't seem to be one particular expected way to handle explicit bases in these corner cases. So I'm fine with not catching the exception there. I've updated the PR.